### PR TITLE
Update to v0.3.6

### PR DESCRIPTION
**Pull request by ukff**
Update to v0.3.6

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -60,5 +72,7 @@ manager:
   securityContext: {}
   health_service:
     enabled: false
+  kubernetesMatchLabels:
+    enabled: false
```
Comment (PK85) [inline chart/values.yaml:76]: we need "true"

File: chart/values.yaml [hunk 2]
```diff
@@ -26,9 +27,20 @@ manager:
...
+    memory_limit: ""
+    cpu_limit: ""
+    req_memory_limit: ""
+    req_cpu_limit: ""
+  imagePullSecrets: []
+#   - name: registrySecretName
+  certificates:
     # Configure if https://github.com/jetstack/cert-manager is used
-    # certManager: true
+    certManager: true
```
Comment (PK85) [inline chart/values.yaml:43]: we need "false" We will use selfsigned certs.

File: chart/values.yaml [hunk 3]
```diff
@@ -26,9 +27,20 @@ manager:
     sm_url: ""
     tokenurl: ""
     tokenurlsuffix: "/oauth/token"
-  certificates: {}
+  rbacProxy:
```
Comment (PK85) [inline chart/values.yaml:30]: this is ok, looks they make it configurable.

File: chart/values.yaml [hunk 4]
```diff
@@ -4,17 +4,18 @@
 
 manager:
   memory_limit: 200Mi
-  cpu_limit: 250m  
+  cpu_limit: 250m
   req_memory_limit: 20Mi
-  req_cpu_limit: 10m
+  req_cpu_limit: 100m
```
Comment (PK85) [inline chart/values.yaml:9]: keep it 10m

File: chart/values.yaml [hunk 5]
```diff
@@ -4,17 +4,18 @@
...
   allow_cluster_access: true
   allowed_namespaces: []
-  replica_count: 1
-  enable_leader_election: false
+  replica_count: 2
+  enable_leader_election: true
```
Comment (PK85) [inline chart/values.yaml:13]: does enable_leader_election to true is related to replica_count 2?

File: chart/templates/webhook.yml [hunk 6]
```diff
@@ -25,18 +25,6 @@ data:
   tls.key: {{ b64enc $cert.Key }}
 ---
 {{- end}}
-{{- if .Values.manager.certificates.selfSigned }}
```
Comment (PK85) [inline chart/templates/webhook.yml:28]: we need that configuration back. @piotrmiskiewicz without that current solution will not work right?

**Review by PK85 — CHANGES_REQUESTED** [review]

File: chart/templates/webhook.yml (hunk 6 above)
Comment (PK85) [reply chart/templates/webhook.yml:28]: Btw from where u got that 0.3.6 webhook.yml? Upstream 0.3.6 webhook yaml: https://github.com/SAP/sap-btp-service-operator/blob/v0.3.6/sapbtp-operator-charts/templates/webhook.yml

File: chart/values.yaml (hunk 5 above)
Comment (wozniakjan) [reply chart/values.yaml:13]: correct, leader election started in replicated HA control-plane and got adopted by controller-runtime (and kubebuilder)
https://github.com/kubernetes-sigs/controller-runtime/blob/35b47fcad8585d12e7a4a4dd66add945a1c86a7a/pkg/leaderelection/doc.go#L18-L22

File: chart/templates/deployment.yml [hunk 7]
```diff
@@ -22,8 +24,9 @@ spec:
         {{- $secretTls := (include (print $.Template.BasePath "/secret-tls.yml") .) -}}
         {{- $configSha := (print $configmap $secret $secretTls) | sha256sum }}
         checksum/config: {{ $configSha }}
-        checksum/config-webhook: {{ include (print $.Template.BasePath "/webhook.yml") . | sha256sum }}
-        sidecar.istio.io/inject: "false"
```
Comment (wozniakjan) [inline chart/templates/deployment.yml:26]: nit: I think we should keep the istio disablement annotation. If users click on installing istio in kyma, btp-operator will stop working which is imho bad UX.

but after upstream merged https://github.com/SAP/sap-btp-service-operator/pull/214#discussion_r1039550062, we can put it to `values.yaml` as recommended below

File: chart/values.yaml [hunk 8]
```diff
@@ -4,17 +4,18 @@
 
 manager:
```
Comment (wozniakjan) [inline chart/values.yaml:5]: continuing the nit with proposed improvement:
```suggestion
manager:
  annotations:
    sidecar.istio.io/inject: "false"
```

File: chart/values.yaml [hunk 9]
```diff
@@ -26,9 +27,20 @@ manager:
...
-  certificates: {}
+  rbacProxy:
+    image:
+      repository: quay.io/brancz/kube-rbac-proxy
+      sha: ""
+      tag: v0.11.0
+    memory_limit: ""
+    cpu_limit: ""
+    req_memory_limit: ""
+    req_cpu_limit: ""
```
Comment (wozniakjan) [inline chart/values.yaml:38]: we should either omit these or set them to some correct values as this renders to invalid resources containing under `deployment.spec.template.spec.containers[kube-rbac-proxy]`
```
...
          resources:
            limits:
            requests:
...
```

Comment (PK85) [reply chart/values.yaml:38]: Lets omit them, as before we had default values for kube-rbac-proxy

**Review by wozniakjan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**